howtos/tech/build-cmocka.rst

```diff
@@ -0,0 +1,135 @@
+.. _build-cmocka:
+
+Build Cmocka for Xtensa
+#######################
+
+Cmocka for SOF is built automatically by default.
+However sometimes you may need to have prebuilt version that can be used with :ref:`CMOCKA_DIRECTORY <cmocka-directory-label>`.
+
+This article exaplains how to build Cmocka manually.
+
+Cmocka fork
+***********
+
+We use our Cmocka fork that adds some options for embedded compilers:
+
+WITH_POSITION_INDEPENDENT_CODE
+   Some compilers cannot compile CMake's test program with PIC,
+   so you can disable it with this option.
+
+WITH_TINY_CONFIG
+   Usually compiler checks are cheap, so configuration is quick.
+   However many compilers are expensive to call,
+   so this option can be used to minimize checks count.
+
+WITH_SHARED_LIB
+   For compilers that cannot build shared libs and still want to
+   use make / make install for making prebuilt libs
+
+All changes made to Cmocka are enabled with these options, without them it works just like vanilla Cmocka.
+
+Clone repo and enter its directory, because all examples will be executed there:
+
+   .. code-block:: bash
+
+      git clone https://github.com/thesofproject/cmocka
+      cd cmocka
+
+Simple build on Linux
+*********************
+
+On any system that CMake identifies as Unix, you can just call following
+commands and it should work:
+
+   .. code-block:: bash
+
+      mkdir build && cd build
+      cmake \
+       -DCMAKE_C_COMPILER=xt-xcc \
+       -DWITH_STATIC_LIB=ON \
+       -DWITH_SHARED_LIB=OFF \
+       -DWITH_EXAMPLES=OFF \
+       -DWITH_POSITION_INDEPENDENT_CODE=OFF \
+       -DCMAKE_INSTALL_PREFIX=install \
+       ..
+      make install
+
+Arguments used:
+
+- CMAKE_C_COMPILER - we specify "xt-xcc" as compiler that CMake should use
```

Comment:
Lines 59-70: I recommend using a definition list here as you did form line 16 above.